jdaviz/configs/imviz/helper.py

```diff
@@ -26,6 +30,65 @@ def load_data(self, data, parser_reference=None, **kwargs):
 
         self.app.load_data(filepath, parser_reference=parser_reference, **kwargs)
 
+    def load_regions(self, regions, data_label, viewer_reference='viewer-1',
+                     **kwargs):
+        """Load a given region into viewer.
+
+        Parameters
+        ----------
+        regions : dict
+            Dictionary mapping desired region names to respective Astropy
+            region objects.
+
+        data_label : str
+            Label to retrieve a specific data set from the viewer instance.
+            Subset from region will be created based on this data.
+
+        viewer_reference : str
+            The reference to the viewer defined with the ``reference`` key
+            in the Imviz YAML configuration file.
+
+        kwargs : dict
+            Extra keywords to be passed into the region's ``to_mask`` method.
+
+        """
+        viewer = self.app.get_viewer(viewer_reference)
+        data = None
+
+        # Cannot use self.app.get_data_from_viewer because we want to bypass
+        # data translator.
+        for layer_state in viewer.state.layers:
+            if (hasattr(layer_state, 'layer') and
+                    layer_state.layer.label == data_label and
+                    isinstance(layer_state.layer, BaseData)):
+                data = layer_state.layer
+                break
+
+        if data is None:
+            raise ValueError('data not found')
+
+        for subset_label, region in regions.items():
+            mask = region.to_mask(**kwargs)
+            im = mask.to_image(data.shape)
+
+            # TODO: This is not registering to GUI. Also breaks get_regions().
+            # ValueError: Several subsets are present, specify which one to retrieve with subset_id=
+            state = ElementSubsetState(indices=np.where((im > 0).flat)[0])
+            data.new_subset(state, label=subset_label)
```

Comment:
@astrofrog , need help here. I got the code to display the regions on display, but it doesn't register to the "subset" menu. I think I got close but somehow still so far.

Also, instead of waiting for upstream fixes for this one, I think we can put in a hacky implementation until upstream fixes are in, because I am not optimistic on fast turnaround time if glue core itself needs to be patched and the MVP status of this feature.

Reply:
You need to call ``DataCollection.new_subset_group`` instead - also note that MaskSubsetState (I think that's what it's called) would be much more efficient than ElementSubsetState

Reply:
Can you please be more specific? How do I access the `DataCollection`? There are a lot of intertwine in the stack that I still don't fully understand.

Reply:
OK, I think I get what you mean. But it still didn't really fix anything, except for now the selection appears in drop-down. I'll update the original post above with a list of things I found so far.